### PR TITLE
test: consolidate test files and reorganize test structure

### PR DESCRIPTION
**Pull request by eunolabs**
## Summary

Consolidates 30+ single-purpose test files into larger, more cohesive test modules organized by functionality. Renames test files to better reflect their scope and merges related test cases to reduce file fragmentation while maintaining full test coverage.

## Why

The test suite had grown to 30+ files with many containing only 50–300 lines of tests for a single feature or code path. This fragmentation made it harder to:
- Locate related tests for a given feature
- Understand the full test surface for a component
- Maintain consistent test patterns across the codebase
- Navigate the test directory structure

Consolidating related tests into larger, logically-grouped files improves maintainability and discoverability without changing test behavior or coverage.

## Type of change

- Refactor or chore (no behavior change)

## How it was tested

```
make test    # All tests pass; coverage unchanged
make lint    # No new findings
```

All existing tests remain unchanged in behavior; this is purely a reorganization of test file structure.

## Security considerations

No security impact — this is a test-only refactor with no changes to production code.

## Checklist

- [x] `make test` passes — race detector clean, `pkg/` coverage stays ≥ 80%
- [x] `make lint` passes — `go vet` + golangci-lint, no new findings
- [x] `make check-license` passes — license header on every new file
- [x] No behavior changes to production code

## Details

**Consolidations in `cmd/eunox-mcp/`:**
- `audit_test.go` — merged from `audit_chain_test.go`, `audit_target_key_test.go`, `audit_coverage_test.go`, `audit_rotate_keys_test.go`, `audit_largeint_test.go`, `audit_copy_test.go`, `audit_close_race_test.go`
- `pdp_jwt_test.go` — merged from `jwt_allowlist_test.go`, `jwt_and_chain_test.go`, `jwt_list_filter_test.go`, `jwt_audience_config_test.go`
- `http_test.go` — merged from `dry_run_test.go`, `coverage_gaps_test.go`
- `main_test.go` — merged from `validate_config_test.go`
- `manifest_test.go` — merged from `manifest_namespace_test.go`, `manifest_ipcompile_test.go`, `manifest_jsonfieldkeys_test.go`, `description_hash_test.go`, `pdp_argument_schema_test.go`, `pdp_directives_test.go`, `pdp_target_claims_test.go`
- `gateway_test.go` — merged from `gateway_schema_test.go`
- `doctor_test.go` — merged from `doctor_live_test.go`
- `drift_test.go` — merged from `drift_coverage_test.go`, `drift_http_test.go`
- `integration_test.go` — merged from `validate_live_integration_test.go`
- `live_upstream_test.go` — merged from `live_upstream_stdio_test.go`
- `transport_unit_test.go` — merged from `proxy_transport_test.go`
- `benchmark_test.go` — merged from `benchmark_hotpath_test.go`
- Removed: `pdp_manifest_test.go`, `unmapped_method_test.go`, `origin_validation_test.go`, `principal_test.go`, `stats_test.go`, `proxy_config_test.go`, `examples_policies_test.go`, `issue_fixes_test.go`, `leak_test.go`

**Consolidations in `pkg/`:**
- `callcounter/memory_internal_test.go` — merged from `cleanup_internal_test.go`, `compact_internal_test.go`, `maxkeys_internal_test.go`, `window_isolation_internal_test.go`
- `capability/condition_test.go` — renamed from `capability_test.

https://claude.ai/code/session_0181DjFvkCcFc6GriKX5dtpS

### CODE REVIEW

File: cmd/eunox-mcp/audit_test.go
Comment (eunolabs) [inline cmd/eunox-mcp/audit_test.go:1]: **Lost file-level doc comments (main review finding).** The merge that produced this file dropped the per-file documentation block that each source file carried between the SPDX header and `package main`. For the 8 files folded in here, that included e.g.:

- `audit_close_race_test.go` — the issue #231 close-race regression rationale ("Close() must not panic with 'send on closed channel'…").
- `audit_only_test.go` — the `--audit` (observe) mode semantics-under-test list.
- `audit_rotate_keys_test.go` — why the rotation/HMAC-key edge-case branches exist.
- `audit_target_key_test.go` — the SPEC §§5.8/7.4 references for agent_id/task_id stamping.

This pattern repeats across ~40 merged files (verified by diffing pre-`package` comment blocks of the deleted/renamed files). The test *functions* are fully preserved, but the explanatory context — issue numbers, acceptance criteria, ADR/SPEC pointers — is gone.

Recommend re-adding a consolidated header to each merged file that stitches the source files' doc comments together. I introduced this in the merge tooling and am happy to push a follow-up that restores them.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: cmd/eunox-mcp/pdp_test.go
Comment (eunolabs) [inline cmd/eunox-mcp/pdp_test.go:1]: Same dropped-doc-comment issue as `audit_test.go` applies here — the headers from `pdp_argument_schema_test.go`, `pdp_directives_test.go`, `pdp_target_claims_test.go`, `principal_test.go`, `coverage_gaps_test.go`, `issue_fixes_test.go` (#326/#329), and `unmapped_method_test.go` (the AUTHORIZATION_FAILED/-32001 acceptance criteria) were lost.

Separately, a softer point: this file now consolidates 8 sources and is the largest in the package. That's defensible (it's all the PDP decision surface), but if it gets unwieldy, a reasonable split would be `pdp_test.go` (Decide/list-filter/redaction) vs `pdp_schema_test.go` (argumentSchema). Non-blocking — flagging only so the consolidation choice is a conscious one.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: pkg/enforcement/enforcement_test.go
Comment (eunolabs) [inline pkg/enforcement/enforcement_test.go:1]: Dropped doc comments here are the most costly of the set because they were almost pure traceability: the folded-in `arch_review_fixes_test.go`, `issue_fixes_test.go`, and `issues_306_316_test.go` headers enumerated the exact issues/arch-review findings each regression guards (#308/#310/#311/#314, #317/#320/#321/#322/#324/#330, and the `docs/arch-review-2026-06-01.md` items). Without those headers the only way to map a test back to its issue is `git log`. Worth restoring as a comment block per merged group.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: .github/workflows/go-ci.yml [hunk 1]
```diff
@@ -142,134 +142,24 @@ jobs:
...
+      #   - conditions/directives split + redactFields obligations and recursion
+      #   - argumentSchema structural validation ordering (INVALID_PARAMS first)
+      #   - JSON-RPC denial codes: exact integers, symbolic message, no raw args
+      #   - unmapped methods denied AUTHORIZATION_FAILED and never forwarded
+      #   - JWT &-chains, exhaustive allowlist, OR-list, and list filtering
+      #   - Rego input contract: arguments{} / directives[] always populated
+      #   - namespace isolation + manifest load-time validation
+      - name: Enforcement invariant gate
+        run: |
+          go test -race -count=1 ./... -run \
```
Comment (eunolabs) [inline .github/workflows/go-ci.yml]: Consolidating the 15 steps into one gate is good (they were redundant with the `test` job). One latent risk worth noting: `go test -run <regex>` exits 0 when the pattern matches **zero** tests. So if any guarded test is later renamed out from under this regex, the gate goes green vacuously instead of failing — the protection silently evaporates. The old per-step form had the same name-coupling, but splitting failures across 15 named steps made a "0 tests matched" easier to spot.

Two low-cost mitigations if you want to keep the signal: (a) add `-v` and grep that an expected count of `--- PASS` lines ran, or (b) drop a one-line comment here telling future editors these prefixes are load-bearing and must track renames. Non-blocking.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by eunolabs — COMMENTED** [review]
## Review: test-suite reorganization

### What this PR does
Renames/merges test files so each unit-test file mirrors its source file (`cmd/eunox-mcp` 64 → 25 files), folds scattered coverage/regression grab-bags into source-named files, consolidates integration tests into `integration_test.go`, and collapses the 15-step `enforcement-compliance` CI job into one invariant gate. Pure reorganization — no production code touched.

### Strengths
- **No test lost.** The set of `Test`/`Benchmark`/`Example` functions is identical before and after (1597, name-diff clean). This is the single most important property for a refactor like this and it holds.
- **Package boundaries respected.** Internal (`package X`) white-box tests were never merged into external (`package X_test`) files; the repo's `*_internal_test.go` convention is preserved (e.g. `engine_internal_test.go`, `memory_internal_test.go`).
- **References updated.** The `Makefile` `check-stale-syntax` allowlist was correctly repointed to the renamed `manifest_test.go` / `capability_test.go`, and `check-stale-syntax`, `gofmt`, license, `go vet`, and `golangci-lint` all pass.
- **Good judgment on the exceptions.** Keeping `adversarial_test.go` and `enforcement_gaps_test.go` as named suites is right — both are referenced by name in `docs/threat-model-mcp.md` / `CONTRIBUTING.md`, so merging them would have churned a security doc.

### Issues
1. **(Should fix) Dropped file-level doc comments.** ~40 merged files lost the documentation block between the SPDX header and `package` — issue numbers, acceptance criteria, ADR/SPEC references. See inline notes on `audit_test.go`, `pdp_test.go`, and `enforcement_test.go`. This is a real documentation regression from the merge tooling; the fix is to re-attach consolidated headers.
2. **(Non-blocking) CI gate can pass vacuously.** `go test -run <regex>` exits 0 on zero matches, so a future test rename could silently drop coverage from the new single gate. See inline note on `go-ci.yml`.

### Trade-offs (acknowledged, not objections)
- A few files remain feature-named rather than source-named (`operational_hardening_test.go`, `low_severity_fixes_test.go`, `transport_unit_test.go`, `gateway_test.go`) because they genuinely span multiple sources. Reasonable given the "ideally" framing of the goal.
- `pdp_test.go` is large (8 sources). Cohesive, but worth a conscious call (inline note).

### Verdict
Mechanically sound and well-verified; safe to merge once the dropped doc comments are restored (issue 1). I can push that follow-up to this branch.

---
_Generated by [Claude Code](https://claude.ai/code)_
